### ISSUE TITLE
undefined symbol when uploading

### ISSUE THREAD

**Issue by bernt-matthias** [galaxyproject/galaxy#4545]
When uploading fasta content via copy-paste some users (windows/firefox) get the following error: 

```
Traceback (most recent call last):
  File "/gpfs1/data/galaxy_server/galaxy-dev/tools/data_source/upload.py", line 9, in <module>
    import gzip
  File "/global/apps/bioinf/galaxy/bin/Python-2.7.13/lib/python2.7/gzip.py", line 10, in <module>
    import io
  File "/global/apps/bioinf/galaxy/bin/Python-2.7.13/lib/python2.7/io.py", line 51, in <module>
    import _io
ImportError: /data/galaxy_server/galaxy-dev/.venv/lib/python2.7/lib-dynload/_io.so: undefined symbol: _PyCodec_LookupTextEncoding
Traceback (most recent call last):
  File "/gpfs1/data/galaxy_server/galaxy-dev/jobs_dir/000/341/set_metadata_CBT3z_.py", line 1, in <module>
    from galaxy_ext.metadata.set_metadata import set_metadata; set_metadata()
  File "/gpfs1/data/galaxy_server/galaxy-dev/lib/galaxy_ext/metadata/set_metadata.py", line 25, in <module>
    import galaxy.model.mapping  # need to load this before we unpickle, in order to setup properties assigned by the mappers
  File "/gpfs1/data/galaxy_server/galaxy-dev/lib/galaxy/__init__.py", line 10, in <module>
    import zipfile
  File "/global/apps/bioinf/galaxy/bin/Python-2.7.13/lib/python2.7/zipfile.py", line 6, in <module>
    import io
  File "/global/apps/bioinf/galaxy/bin/Python-2.7.13/lib/python2.7/io.py", line 51, in <module>
    import _io
ImportError: /data/galaxy_server/galaxy-dev/.venv/lib/python2.7/lib-dynload/_io.so: undefined symbol: _PyCodec_LookupTextEncoding
```

They copy-pasta the examples from http://rna.informatik.uni-freiburg.de/IntaRNA/Input.jsp?example=IntaRNA_3189420 

In the shared object the symbol is indeed undefined 

```
nm /data/galaxy_server/galaxy-dev/.venv/lib/python2.7/lib-dynload/_io.so | grep LookupTextEncoding
                 U _PyCodec_LookupTextEncoding

```

**Comment by nsoranzo** [comment]
I think this may be caused by using two different Python versions: the one in `/global/apps/bioinf/galaxy/bin/Python-2.7.13/` and the one in `/data/galaxy_server/galaxy-dev/.venv/` .

**Comment by bernt-matthias** [comment]
Thanks for the reply: 

The two python versions are actually intended (I think/hope): The venv at `/data/galaxy_server/galaxy-dev/.venv/` is based on the python in `/global/apps/bioinf/galaxy/bin/Python-2.7.13/`. 

The python at `/global/apps/bioinf/galaxy/bin/Python-2.7.13/` is needed for the python shared libraries, because the virtualenv only contains the python binaries but not the python libraries. Thats something I never understood (my expectation was that a venv should contain everything to run python). 

But the actual problem could be related and your comment brought me on the right track. Since our galaxy runs jobs as real user the user configs (bashrc,..) could load other python versions which might confuse the system. I will test this and report here.

**Comment by bernt-matthias** [comment]
Got it sorted out. The problem was that one user had an entry in his bashrc that loaded a different python version (we got a module system running here). So I created a small shell scipt that purges all loaded python modules and loads the one that is the bases of galaxy's virtualenv.  This shell script is now used in all jobs by addin `<env file="/gpfs1/data/galaxy_server/galaxy-dev/config/job_setup_user_env.sh" />` to each destination.

Thanks for the hint.
